### DISCUSSION TITLE
Loading 2 unique subs for various game modes

### DISCUSSION THREAD

**Discussion by DanJF** [FakeFishGames/Barotrauma#5555] [Ideas and suggestions]
Current state : 
Only PvP supports game-driven loading of two subs, however they are only ever mirrored/duplicate subs.
This is highly limiting not only to PvP, but misses enormous potential in other game modes for players to interact and create/undergo unique encounters which do not necessarily have to be antagonistic.

Off-the-hip, here are examples of how each game mode could benefit from the game facilitating loading of 2 unique subs in a given mission.

### Campaign:
- Encounters with submarines of other factions (assuming the Ai can be brought up to a fundamentally functioning capacity). This would add immersion to campaign, with the player's experience of factions not being strictly limited to person-to-person encounters on stations. The Jovians, Church of the Husk, and Clowns of the Honk Mother, are _supposedly_ a threat, and yet none of them threaten the players in open waters where the overwhelming majority of the game takes place. The Jovians/Renegades are given their own distinct Captain sub-class, and yet nobody has ever been threatened by a Jovian/Renegade sub in the oceans of Europa, except for the PvP game mode. It's incredibly inconsistent in nature/gameplay with the lore of the setting.

### Mission
- The primary experience of Barotrauma is driven by player-to-player interactions, and currently it is limited/restricted to players interacting entirely within the same sub. There are multitudes of missions, antagonistic _and_ cooperative, which would greatly benefit from having distinct separate subs loaded into the game from either station. Interdiction missions, searching for contraband which may or may not be on a sub. Humanitarian missions, in which one sub may start downed and damaged on the sea floor, and the other sub must come to rescue and evacuate them to a station. Coalition vs Jovians, with Jovians perhaps trying to pose as a civilian vessel trying to get by, while smuggling goods, or on a mission to capture a Coalition submarine for their own use. Civilian cruise ships set to observe "the dangerous depths of Europa", only to be confronted by a Church of the Husk vessel seeking new members. The possibilities are limited only by the imagination.

### Sub v Sub
- This one is obvious, but arguably just as important. This game has a **huge**, multi-national following. Russia, China, Japan, Europa, United Kingdom, Austraila, United States, South America, and many more countries all have player bases which play Barotrauma. The game mode is primed for hosting friendly international competition, but having _every_ match be a mirrored scenario is incredibly limiting. This mode would potentially explode if teams were allowed to have their own distinct subs, and would generate an incredible volume of dialogue around sub building. Which Vanilla sub is superior? Who would win in a lineup of EMS vs EK? Without any intervention of balancing from Devs, we could possibly see a community-driven competitive scene emerge as well. But, that's not the goal. The real goal is just more interesting and engaging gameplay, and letting players smash their subs against each other like little kids having fun.

### tl:dr ;
- All modes of Barotrauma would benefit from this implementation. It's lightning in a bottle, and I hope you'll consider it before dismissing it. Supposedly, the implementation for such a thing already exists within admin commands (which aren't listed/detailed with the help feature), so it could potentially be "easy" to implement in some modes.

Thanks for reading, and thanks for your work on the game thus far.
Cheers.
-tons0phun
